modules/stochastic_tools/src/reporters/MappingReporter.C

```diff
@@ -0,0 +1,224 @@
+  // there is a variable in the system and if found, just reduce it.
+  if (_parallel_storage)
+  {
+    // If the mapping is not built yet, we shall build it using the solutions in the parallel
+    // storage. The conditional process is decided in the mapping object.
+    for (const auto & var : _variable_names)
+      _mapping->buildMapping(var);
+
+    // Since the solution fields can be distributed among the processes of each sub-application
+    // (unlike samples and reporter values which are just on the root process), we need to use the
+    // information in the sampler to check which solution is where.
+    const auto rank_config = _sampler->getRankConfig(true);
+
+    // We need to do some parallel communication in case we have snapshots on processes other than
+    // the roots of the sub-applications. This will collect the vectors that we need to send in the
+    // following format:
+    // <to which processor, (variable index, global sample index, solution field)>
+    std::unordered_map<processor_id_type,
+                       std::vector<std::tuple<unsigned int, unsigned int, std::vector<Real>>>>
+        send_map;
+
+    // We need to use the rank config here to ensure we are looping on the non-root processors
+    // too
+    for (const auto sample_i : make_range(rank_config.num_local_sims))
+    {
+      std::vector<Real> data = _sampler->getNextLocalRow();
+
+      // Converting the local indexing to global sample indices
+      const unsigned int global_i = sample_i + _sampler->getLocalRowBegin();
+
+      for (const auto var_i : index_range(_variable_names))
+      {
+        // Create a temporary storage for the coordinates in the latent space
+        std::vector<Real> local_vector;
+
+        // Check if the current process has this global sample
+        if (_parallel_storage->hasGlobalSample(global_i, _variable_names[var_i]))
+        {
+          // Fetch the solution vector for the given sample index and variable
+          const auto & full_vector =
+              _parallel_storage->getGlobalSample(global_i, _variable_names[var_i]);
+
+          // At the moment we only support simulations which have only one solution field
+          // per sample. This is typically a steady-state simulation.
+          if (full_vector.size() != 1)
+            mooseError("MappingReporter is only supported for simulations with one solution "
+                       "field per run!");
+
+          // We use the mapping object to generate the coordinates in the latent space
+          _mapping->map(_variable_names[var_i], global_i, local_vector);
+
+          // If we are on the root processor of the sub-application, we simply insert the result
+          // into the reporter storage space. Othervise we will send it to the root process.
+          if (rank_config.is_first_local_rank)
+            (*_vector_real_values_parallel_storage[var_i])[sample_i] = local_vector;
```

Comment:
I'm missing how this works - why do you not need to get the memory address for this to work

Reply:
We discussed this, resolving.